app/src/main/java/com/kodeco/android/countryinfo/ui/screens/countryinfo/CountryInfoViewModel.kt

```diff
@@ -17,53 +21,95 @@ sealed class UiState {
     data class Error(val throwable: Throwable) : UiState()
 
     data object Loading : UiState()
+
+    data object Initial : UiState()
 }
 
 // Intent
 sealed class CountryInfoIntent {
-    object LoadCountries : CountryInfoIntent()
+    data object LoadCountries : CountryInfoIntent()
+    data object LoadFavoriteCountries : CountryInfoIntent()
+    data object LoadCountriesFromLocalStorage : CountryInfoIntent()
     data class FavoriteCountry(val country: Country) : CountryInfoIntent()
+    data class LikeAndRefresh(val country: Country) : CountryInfoIntent()
 }
 
 class CountryInfoViewModel(
-    private val repository: CountryRepository
+    private val countryRepository: CountryRepository,
+    countryPrefs: CountryPrefs
 ) : ViewModel() {
 
-    private val _uiState = MutableStateFlow<UiState>(UiState.Loading)
+    private val _uiState = MutableStateFlow<UiState>(UiState.Initial)
     val uiState = _uiState.asStateFlow()
 
-    init {
-        processIntent(CountryInfoIntent.LoadCountries)
-    }
+    val featureEnabled = countryPrefs.getFavoritesFeatureEnabled()
+
+    val favoriteEnabled = countryPrefs.getFavoriteCountriesFeatureEnabled().stateIn(
+        viewModelScope,
+        initialValue = false,
+        started = SharingStarted.WhileSubscribed(1000)
+    )
+
+    val offlineEnabled = countryPrefs.getLocalStorageEnabled().stateIn(
+        viewModelScope,
+        initialValue = false,
+        started = SharingStarted.WhileSubscribed(1000)
+    )
```

Comment:
Continuing off of what I mentioned about too much logic in the composable view, you could add an init block that listens for changes on the settings prefs values:

```kotlin
    init {
        viewModelScope.launch {
            // drop the first value, as we will already be loading the 
            // countries the first time down at the bottom of this init block
            featureEnabled.drop(1).collect {
                processIntent(CountryInfoIntent.LoadCountries)
            }
        }
        viewModelScope.launch {
            favoriteEnabled.drop(1).collect {
                processIntent(CountryInfoIntent.LoadCountries)
            }
        }
        viewModelScope.launch {
            offlineEnabled.drop(1).collect {
                processIntent(CountryInfoIntent.LoadCountries)
            }
        }

        viewModelScope.launch {
            loadCountries()
        }
    }
```